File: services/veilkey-cli/state.go
package main

import (
	"fmt"
	"os"
	"strings"
)

const (
	pasteModeOn  = "on"
	pasteModeOff = "off"
)

func stateDir() string {
	return getEnv("VEILKEY_STATE_DIR", defaultStateDir())
}

func pasteModePath() string {
	return stateDir() + "/paste-mode"
}

func currentPasteMode() string {
	data, err := os.ReadFile(pasteModePath())
	if err != nil {
		return pasteModeOn
	}
	mode := strings.ToLower(strings.TrimSpace(string(data)))
	if mode == pasteModeOff {
		return pasteModeOff
	}
	return pasteModeOn
}

func pasteTempIssuanceEnabled() bool {
	return currentPasteMode() == pasteModeOn
}

func setPasteMode(mode string) error {
	mode = strings.ToLower(strings.TrimSpace(mode))
	if mode != pasteModeOn && mode != pasteModeOff {
		return fmt.Errorf("invalid paste mode: %s", mode)
	}
	if err := os.MkdirAll(stateDir(), 0o755); err != nil {
		return err
	}
	return os.WriteFile(pasteModePath(), []byte(mode+"\n"), 0o644)
}

File: services/veil-cli/src/state.rs
use std::env;
use std::fs;
use std::path::PathBuf;

pub fn default_state_dir() -> PathBuf {
    let tmp =
        env::var("TMPDIR").unwrap_or_else(|_| std::env::temp_dir().to_string_lossy().into_owned());
    PathBuf::from(tmp).join("veilkey-cli")
}

pub fn state_dir() -> PathBuf {
    match env::var("VEILKEY_STATE_DIR") {
        Ok(v) if !v.is_empty() => PathBuf::from(v),
        _ => default_state_dir(),
    }
}

pub fn paste_mode_path() -> PathBuf {
    state_dir().join("paste-mode")
}

pub fn current_paste_mode() -> &'static str {
    match fs::read_to_string(paste_mode_path()) {
        Ok(data) => {
            if data.trim().to_lowercase() == "off" {
                "off"
            } else {
                "on"
            }
        }
        Err(_) => "on",
    }
}

pub fn set_paste_mode(mode: &str) -> Result<(), String> {
    let mode = mode.trim().to_lowercase();
    if mode != "on" && mode != "off" {
        return Err(format!("invalid paste mode: {}", mode));
    }
    let dir = state_dir();
    fs::create_dir_all(&dir).map_err(|e| e.to_string())?;
    fs::write(paste_mode_path(), format!("{}\n", mode)).map_err(|e| e.to_string())
}
